### PR TITLE
task was done

### PR DESCRIPTION
**Pull request by anton-chornobai**

### CODE REVIEW

**Comment by anton-chornobai** [comment]
[DEMO LINK](https://anton-chornobai.github.io/react_keyboard/) and add it to the PR description.

**Comment by anton-chornobai** [comment]
Idk why tests did fail.

File: src/App.tsx [hunk 1]
```diff
@@ -1,7 +1,31 @@
...
+  };
+
+  componentDidMount() {
+    document.addEventListener('keyup', (event: KeyboardEvent) => {
+      this.setState({ message: `The last pressed key is [${event.key}]` });
+    });
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('keyup', (event: KeyboardEvent) => {
```
Comment (etojeDenys) [inline src/App.tsx:15]: The event listener removal is not working correctly. You should create a separate method for the event handling and use it for adding and removing the listener.

**Review by etojeDenys — CHANGES_REQUESTED** [review]

**Comment by etojeDenys** [comment]
> [DEMO LINK](https://anton-chornobai.github.io/react_keyboard/) and add it to the PR description.

your link does not work

**Pull request closed without merging**